SmartDeviceLink/SDLCacheFileManager.m

```diff
@@ -0,0 +1,282 @@
+        if (!isNewDirectory) {
+            [self sdl_clearLockScreenDirectory];
+        }
+        iconArchiveFile = [[SDLIconArchiveFile alloc] init];
+    } else {
+        for (SDLLockScreenIconCache *iconCache in iconArchiveFile.lockScreenIconCaches) {
+            // The icon exists and is not expired
+            if ([iconCache.iconUrl isEqualToString:request.url] && [self.class numberOfDaysFromDateCreated:iconCache.lastModifiedDate] < 30) {
+                
+                NSString *imageFilePath = [self.sdl_cacheFileBaseDirectory stringByAppendingPathComponent:[self.class sdl_md5HashFromString:request.url]];
+                UIImage *icon = [UIImage imageWithContentsOfFile:imageFilePath];
+                
+                // If we fail to get icon at path, break loop early to download new icon
+                if (icon == nil) {
+                    [self sdl_clearLockScreenDirectory];
+                    iconArchiveFile = [[SDLIconArchiveFile alloc] init];
+                    break;
+                }
+                
+                return completion(icon, nil);
+            }
+        }
+    }
+    
+    // Download Icon
+    [self sdl_downloadIconFromRequestURL:request.url withCompletionHandler:^(UIImage * _Nullable image, NSError * _Nullable error) {
+        if (error != nil) {
+            return completion (nil, error);
+        }
+        
+        // Save lock screen icon to file path
+        NSString *iconFilePath = [self sdl_writeImage:image toFileFromURL:request.url];
+
+        if (iconFilePath == nil) {
+            SDLLogE(@"Could not save lockscreen icon to path: %@", error);
+        }
+        
+        // Update archive file with icon
+        [self updateArchiveFileFromRequest:request iconFilePath:iconFilePath archiveFile:iconArchiveFile];
+        
+        return completion(image, nil);
+    }];
+}
+
+#pragma mark - Cache Helper Methods
+
+/**
+ * Handles request to retrieve lock screen archive file from specified path.
+ *
+ * @param path The directory where the archive file is held
+ * @return The archive file at path
+*/
+- (nullable SDLIconArchiveFile *)sdl_retrieveArchiveFileFromPath:(NSString *)path {
+    NSString *archiveObjectPath = self.sdl_archiveFileDirectory;
+    SDLIconArchiveFile *iconArchiveFile = [NSKeyedUnarchiver unarchiveObjectWithFile:archiveObjectPath];
+    
+    return iconArchiveFile;
+}
+
+/**
+ * Handles the request to clear all files from lock screen icon directory in cache.
+*/
+- (void)sdl_clearLockScreenDirectory {
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    NSString *directory = self.sdl_cacheFileBaseDirectory;
+    NSError *error = nil;
+    
+    for (NSString *file in [fileManager contentsOfDirectoryAtPath:directory error:&error]) {
+        if (error != nil) {
+            SDLLogE(@"Failed retrieving contents of directory: %@ with error: %@", directory, error);
```

Comment:
This should break, right? This method should possibly return an error for this failure since we can't even iterate on the directory.

Reply:
I added the break, however should we not return an error here since this is being used within `retrieveImageForRequest:withCompletionHandler`? I don't believe we would do anything with the error inside that method since we'd still like to download the icon.

Reply:
I think that's right, see my previous response.